src/Sylius/Bundle/ProductBundle/DependencyInjection/SyliusProductExtension.php

```diff
@@ -166,4 +170,50 @@ private function prependVariation(ContainerBuilder $container, array $config)
             ],
         ]);
     }
+
+    /**
+     * Add elastica product listener
+     *
+     * Only if elastica is search engine
+     *
+     * @param ContainerBuilder $container
+     */
+    public function prependElasticaProductListener(ContainerBuilder $container)
+    {
+        $config = $this->processConfiguration(new Configuration(), $container->getExtensionConfig($this->getAlias()));
+
+        if (!$container->hasExtension('fos_elastica') || !$container->hasExtension('sylius_search')) {
+            return;
+        }
+
+        $syliusSearchConfigs = $container->getExtensionConfig('sylius_search');
+
+        // Get engine
+        $engine = null;
+        foreach ($syliusSearchConfigs as $syliusSearchConfig) {
+            if (isset($syliusSearchConfig['engine'])) {
+                $engine = $syliusSearchConfig['engine'];
```

Comment:
Shouldn't you add `break;` in this `if`? Cause now last one wins, with `break` first one wins.